spec/pb_kits/playbook/kits/source_spec.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+require_relative "../../../../app/pb_kits/playbook/pb_source/source"
+
+RSpec.describe Playbook::PbSource::Source do
+  subject { Playbook::PbSource::Source }
+
+  it { is_expected.to define_boolean_prop(:hide_icon)
+                      .with_default(false) }
+  it { is_expected.to define_prop(:source).of_type(Playbook::Props::String) }
+  it { is_expected.to define_enum_prop(:type)
+                      .with_values("user", "retail", "inbound", "outbound", "prospecting", "events", "referral")
+                      .with_default("inbound") }
+  it { is_expected.to define_prop(:user).of_type(Playbook::Props::Hash)
+                      .with_default({}) }
+
+  describe "#type_text" do
+    it "titlizes the type when no user is present" do
+      expect(subject.new(type: "prospecting").type_text).to eq "Prospecting"
+    end
+
+    it "returns the user's name when a user is present" do
+      expect(subject.new(type: "user", user: {name: "New User"}).type_text).to eq "New User"
+    end
+  end
+
+  describe "#user_id" do
+    it "returns the user_id if a user is present with a user id, or returns nil", :aggregate_failures do
+      expect(subject.new(type: "prospecting", user: {}).user_id).to eq nil
+      expect(subject.new(type: "referral", user: { name: "User" }).user_id).to eq nil
+      expect(subject.new(type: "user", user: { name: "User", user_id: 435 }).user_id).to eq 435
+
+    end
+  end
+
+  describe "#show_icon?" do
```

Comment:
I love all the extra tests you put in here! Nice work!

Reply:
Thanks!